src/main/java/com/example/model/CartItem.java

```diff
@@ -18,7 +18,7 @@
 
 @Getter
 @Setter
-@EqualsAndHashCode(of = "id")
+@EqualsAndHashCode(of = {"shoppingCart", "book"})
 @ToString
 @SQLDelete(sql = "UPDATE cart_items SET is_deleted = true WHERE id=?")
 @Where(clause = "is_deleted = false")
```

Comment:
Use @SqlRestriction instead of @Where. @Where is depracated.

Reply:
done, thank you